lib/writer.js

```diff
@@ -153,7 +144,7 @@
     };
     function callback(it) {
       const ret = extended_attributes(it.extAttrs);
-      return `${ret}callback ${it.name} = ${type(it.idlType)}(${args(it.arguments)});`;
+      return `${ret}callback ${it.name} = ${type(it.idlType)}(${it.arguments.map(argument).join(",")});`;
```

Comment:
Nit: I wonder if we should consider splitting these up a little bit, because any more complexity is going to make these hard to read and maintain... could be:

```JS
function callback(it) {
  const { extAttrs, name, arguments:args, idlType } = it; // "it" could have a better name too
  const extendedAttrs = extended_attributes(extAttrs);
  const type = type(it.idlType);
  const arguments = args.map(argument).join(",");
`${extendedAttrs}callback ${name} = ${type}(${arguments});`
```

Reply:
A triple 👍 for better readability.

Reply:
But how should the names of local variables be?

1. `const extendedAttrs = extended_attributes(extAttrs);` is bad because camelCase and underscore is mixed
2. `const type = type(it.idlType);` has a name collision.
3. `const arguments = args.map(argument).join(",");` is invalid in strict mode

IMO these are not just simple mistakes but show it's hard to name them.

Reply:
Agree, naming sucks :( Why this was just a nit. Let's go with this for now but maybe come back to it if it becomes a problem.

Reply:
This definitely is a problem in #185, hmm.